tests/e2e/framework/rawVM.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	u "istio.io/istio/tests/e2e/util"
+)
+
+// RawVM interfaces different cloud venders to support e2e testing with VM
+type RawVM interface {
```

Comment:
Is there a way to simplify this ? Can we request the provider to set ssh keys and then use common ssh libraries to interact with the VM ? ie https://godoc.org/golang.org/x/crypto/ssh
All you would need to provide is an IP and a path to ssh key.

Reply:
I actually think calling `gcloud` is simpler and safer. It also handles secret distribution where the test runs in a new pod which is considered as a new trusted entity and gcloud mints a new key pair at run time.

Reply:
yes it's fine to depend on gcloud for now

Reply:
ok.